neutrons/reflectivity_ui
# Add option to use y pixel range from data run also for the direct beam

## Description of work:

Add an option (checkbox in the UI) to use the `low_res_roi` , or `y` pixel range, from the data set also for the direct beam when normalizing. The default is to use the `low_res_roi` from the direct beam table as before.

# Manual test for the reviewer
([instructions to set up the environment](https://reflectivity-ui.readthedocs.io/en/latest/developer/environment.html))

In the source repo directory with conda environment activated:
1. `pip install .  # install reflectivity_ui package`
2. `bin/quicknxs2  # start QuickNXS`
3. Click toolbar menu `File > Open File` and open `<source dir>/test/data/reflectivity_ui-data/REF_M_42099.nxs.h5`
4. Select peak region and add to direct beam table using button `I(lambda)`
5. In the file list, select `REF_M_42112.nxs.h5`, select peak region and add to the reduction table using plus button
6. Check the checkbox `Lock direct beam y0 and yw to data run` and verify that the reflectivity plot is updated.
![image](https://github.com/user-attachments/assets/d95905df-6c84-4e82-982a-0f2ff54e5a9e)
7. Click button `Reduce...` and `Ok` and verify that the reduction is run as expected.
8. Also test that the state of the chechbox is updated correctly when loading a reduced data file.

# Check list for the reviewer
or an explanation is provided as to why release notes are unnecessary
